### PR TITLE
Fix import bug

### PR DESCRIPTION
**Pull request by donfrigo**
### Description of the Change
Restored the functionality of the smart contract import modal.

### Benefits
It is functional again.

### Verification Process
In the explorer right click on any folder to bring up the file import modal.

### CODE REVIEW

File: src/reducers/utils/fileUtils.ts [hunk 1]
```diff
@@ -27,3 +28,33 @@ export function isValidProjectItemName(name: string): boolean {
     name = name.trim();
     return name.length > 0 && name.length < 255 && !UNIX_RESERVED_CHARS_REGEX.test(name) && !WINDOWS_RESERVED_CHARS_REGEX.test(name);
 }
+
+export function generateItem(name: string, type: ProjectItemTypes, code?: string): IProjectItem {
```
[the remark this answers is not in the document]
Comment (donfrigo) [reply src/reducers/utils/fileUtils.ts]: fixed

File: src/reducers/explorer.reducer.ts [hunk 2]
```diff
@@ -232,23 +232,16 @@ export default function explorerReducer(state = initialState, action: AnyAction)
...
-                .map((path: string) => path.startsWith('/') ? path : '/' + path)
-                .map((path: any) => path.split('/').slice(1))
-                .reduce((children: any, path: any, idx: number) => insert(children, path, importSourceArray[idx]), <IProjectItem[]>[]);
-
             // add new item to the tree
             const [newTree, replacedTargetItem] = updateItemInTree(
                 state.tree,
                 parentId,
-                i => ({ ...i, children: sortProjectItems(i.children.concat(objectArray)) })
+                i => ({ ...i, children: sortProjectItems(i.children.concat(items)) })
```
Comment (ostap1010) [inline src/reducers/explorer.reducer.ts]: There is one bottleneck here - what happens if item with same name already exist? 
I'd say where we replace existing items or show a warning that it should be deleted.

File: src/reducers/explorer.reducer.ts [hunk 3]
```diff
@@ -232,23 +232,16 @@ export default function explorerReducer(state = initialState, action: AnyAction)
...
             );
 
             // parent item was found and child was added
             if (replacedTargetItem) {
-                objectArray.forEach((item) => {
+                items.forEach((item: any) => {
```
Comment (ostap1010) [inline src/reducers/explorer.reducer.ts]: This is very weird loop - it is just reassigning itemNameValidation to same object in every cycle. Is it needed at all?

File: src/reducers/explorer.reducer.ts (hunk 2 above)
Comment (donfrigo) [reply src/reducers/explorer.reducer.ts]: items with the same name are replaced now

File: src/reducers/explorer.reducer.ts (hunk 3 above)
Comment (donfrigo) [reply src/reducers/explorer.reducer.ts]: removed

File: src/reducers/utils/fileUtils.ts [hunk 4]
```diff
@@ -46,3 +46,12 @@ export function insert(children: any[], [head, ...tail]: any, code: string): IPr
     }
     return children;
 }
+
+// Insert item into tree structure overwriting items with same type and name
+export function appendWithoutDuplicate(baseArray: IProjectItem[], newArray: IProjectItem[]): IProjectItem[] {
```
Comment (ostap1010) [inline src/reducers/utils/fileUtils.ts]: Sorting can be performed in this function, cause there would be no case when we need not sorted array.

File: src/reducers/utils/fileUtils.ts [hunk 5]
```diff
@@ -46,3 +46,12 @@ export function insert(children: any[], [head, ...tail]: any, code: string): IPr
     }
     return children;
 }
+
+// Insert item into tree structure overwriting items with same type and name
```
Comment (ostap1010) [inline src/reducers/utils/fileUtils.ts]: Please use javadoc comments style

File: src/reducers/utils/fileUtils.ts (hunk 4 above)
Comment (donfrigo) [reply src/reducers/utils/fileUtils.ts]: Moved sortProject to appendWithoutDuplicate

File: src/reducers/utils/fileUtils.ts (hunk 5 above)
Comment (donfrigo) [reply src/reducers/utils/fileUtils.ts]: fixed

**Pull request merged**